### ISSUE TITLE
[rush] A simpler output directory structure

### ISSUE THREAD

**Issue by liyikun** [microsoft/rushstack#3150]
# Overview
Currently, the output directory of rush deploy is exactly the same as the package structure.

just like

```
<root>
-- <apps>
---- pkg1
---- pkg2
-- <utils>
---- <bussiness1>
------ pkg3
---- <bussiness2>
------ pkg4
...other
```

if pkg1 dependence pkg3 and pkg4

when rush deploy pkg1, it output structure is

> <> is folder, no <> is file

```
<root>
-- <output>
---- <apps>
------ <pkg1>
---- <utils>
------ <bussiness1>
--------- <pkg3>
---- <bussiness2>
--------- <pkg4>
---- <common>
-- deploy-metadata.json
```

I wish there was a better structure

```
<root>
-- <output>
------ <.rush-common>
--------- current common
--------- <dependences>
----------- pkg3 & pkg4
---- pkg1's all code
---- pkg1 node_modules

```

This way we can set up a common startup script for the entire monorepo repository instead of relying on the package path.
like `start.sh`,  it have included in all app pkg root. In this way, the deployment platform set in the basic cloud can do the same settings.

**Comment by liyikun** [comment]
i hope this can be an option for the deploy output, and I'd like to contribute code if possible.

**Comment by liyikun** [comment]
Of course, the reverse effect is to destroy the original package and package directory structure, but I think packages and packages should not have such filepath dependencies directly. should access  package-json.main.

**Comment by LPegasus** [comment]
I think current structure is OK. Keeping output folder tree just like repo has many benefits. No extra knowledge when trouble shooting. Easy to change your code on the server.

**Comment by liyikun** [comment]
@LPegasus  You've hit the nail on the head. I think this feature can be achieved through plugins
